### ISSUE TITLE
The NaNoFuzz button doesn't appear when there are import statements in the file

### ISSUE THREAD

**Issue by ZiyaoWei** [nanofuzz/nanofuzz#169]
Could be me doing something wrong but the `NaNoFuzz...` button only appears when there are no import statements in the file.

Would be happy to look into what's going on if you are fine with having external contributors! Also feel free to point me to other good starter issues, this seems like a really cool project 😄

**Comment by cmumatt** [comment]
Yes, we are really excited about the project and are glad that you think it is cool, too! 🤓

NaNofuzz follows imports, but we have not yet implemented support for the following in NaNofuzz:
- Deconstructed types, OR types, Tuples, Enums, Generics, or Function types
- Non-finite numeric inputs (`NaN`, `Infinity`), the new `bigint` type, and `null`

If the NaNofuzz button does not appear, it's often due to encountering one of the above in one of the function's input type definitions as NaNofuzz tries to resolve the input types, e.g., by resolving imports. There is no particular reason why these could not be supported in the future & community contributions to expand NaNofuzz' support for these are very welcome! 

If you can point to a minimal example in a public repo, we can see if that is what is going on in this case and maybe make a suggestion. Or maybe we find a bug! NaNofuzz outputs some debug into to VSC's console when it runs into an unsupported type definition, but this feedback presently can be pretty cryptic. 

Also, I marked a couple issues as `[good-first-issue](https://github.com/nanofuzz/nanofuzz/labels/good%20first%20issue)`---let me know if you are interested in any of those. We need to create a contributor's guide... ours will probably be similar to the one we use for Penrose: https://github.com/penrose/penrose/blob/main/CONTRIBUTING.md#contributing

**Comment by cmumatt** [comment]
Awesome!

**Comment by cmumatt** [comment]
Closing this question issue for now.
